File: vision/templates.js
// vision/templates.js
import cv from 'opencv4nodejs';
import fs from 'node:fs';
import path from 'node:path';

export function loadTemplates(dir) {
  const files = fs.readdirSync(dir).filter(f => f.endsWith('.png'));
  return files.map(f => ({
    name: f.replace(/(\.png)+$/i, ''),
    mat: cv.imdecode(fs.readFileSync(path.join(dir, f)))
  }));
}

export const rankTemplates = loadTemplates('vision/templates/ranks');
export const suitTemplates = loadTemplates('vision/templates/suits');

File: vision/templates.py
# vision/templates.py
import cv2 as cv
import os


def load_templates(directory):
    files = [f for f in os.listdir(directory) if f.endswith('.png')]
    templates = []
    for f in files:
        path = os.path.join(directory, f)
        mat = cv.imread(path)
        name = f
        while name.lower().endswith('.png'):
            name = name[: -4]
        templates.append({"name": name, "mat": mat})
    return templates

base_dir = os.path.dirname(__file__)
rank_templates = load_templates(os.path.join(base_dir, 'templates', 'ranks'))
suit_templates = load_templates(os.path.join(base_dir, 'templates', 'suits'))
